File: 2025/go/puzzles/dec01.go
package puzzles

import (
	"aoc/utils"
	"fmt"
)

var Dec01 = Puzzle{
	SolvePart1: func(date string, useTestData bool) {
		fmt.Println("Day 01, Part 1")

		const N = 100
		pos := 50
		zeros := 0

		lines, err := utils.DataFileReader(date, useTestData, "1", false)
		if err != nil {
			fmt.Println("Error reading file:", err)
			return
		}

		for _, line := range lines {
			dir := line[0]
			var dist int
			fmt.Sscanf(line[1:], "%d", &dist)

			if dir == 'L' {
				pos = (pos - dist) % N
			} else {
				pos = (pos + dist) % N
			}

			if pos < 0 {
				pos += N
			}

			if pos == 0 {
				zeros++
			}
		}

		fmt.Println("zeros:", zeros)

	}, // SolvePart1
	SolvePart2: func(date string, useTestData bool) {
		fmt.Println("Day 01, Part 2")

		const N = 100
		pos := 50
		zeros := 0

		lines, err := utils.DataFileReader(date, useTestData, "1", false)
		if err != nil {
			fmt.Println("Error reading file:", err)
			return
		}

		for _, line := range lines {
			dir := line[0]
			var dist int
			fmt.Sscanf(line[1:], "%d", &dist)

			for i := 0; i < dist; i++ {
				if dir == 'L' {
					pos = (pos - 1 + N) % N
				} else {
					pos = (pos + 1) % N
				}
				if pos == 0 {
					zeros++
				}
			}
		}

		fmt.Println("zeros:", zeros)
	}, // SolvePart2
}

File: 2025/python/puzzles/dec01.py
from typing import List
from utils.file_utils import data_file_reader



def solve_part1(date: str, use_test_data: bool = False):
	print("Day 01, Part 1")

	lines = data_file_reader(date = date, use_test_data = use_test_data, part = 1)

	N = 100
	pos = 50
	zeros = 0

	for line in lines:
		dir = line[0]
		dist = int(line[1:])
		if dir == 'L':
			pos = (pos - dist) % N
		elif dir == 'R':
			pos = (pos + dist) % N

		if pos < 0:
			pos += N

		if pos == 0:
			zeros += 1

	print(f"Zeros: {zeros}")



def solve_part2(date, use_test_data = False):
	print("Day 01, Part 2")

	lines = data_file_reader(date = date, use_test_data = use_test_data, part = 1)

	N = 100
	pos = 50
	zeros = 0

	for line in lines:
		dir = line[0]
		dist = int(line[1:])
		for i in range (dist):
			if dir == 'L':
				pos = (pos - 1 + N) % N
			elif dir == 'R':
				pos = (pos + 1) % N
			if pos == 0:
				zeros += 1

	print(f"Zeros: {zeros}")
